File: Python/rbf.py
#  FUNCTION File: Radial Basis Function
#  Author: Arshad Afzal, IIT Kanpur, India
#  For Questions/ Comments, please email to arshad.afzal@gmail.com

import numpy as np
from math import *


def rbf(x, c, theta):
    f = np.exp(- pow(theta, 2) * np.sum(np.square(x - c)))
    return f

File: MATLAB/rbf.m
function y = rbf(x,c,theta)

%RBF Gaussian radial basis function
%
%   y = rbf(x, c, theta)
%
%   Inputs:
%       x     : Input vector
%       c     : Center of the radial basis function
%       theta : Shape parameter
%
%   Output:
%       y     : RBF response
         
  y = exp(- theta .^2 * sum((x - c).^2));

end

